lib/src/Base/Stat/SamplePartition.cxx

```diff
@@ -0,0 +1,265 @@
+      {
+        const UnsignedInteger clusterEnd = std::max(clusterStart + 1, j + 1 - streak);
+        Indices selection(clusterEnd - clusterStart);
+        selection.fill(clusterStart);
+        const Point peak = sample_.select(selection).getMax();
+        // keep only clusters with values above threshold
+        if (peak[0] > threshold)
+        {
+          peaks.add(peak);
+          indicesCollection.add(Indices{clusterStart, clusterEnd});
+        }
+        clusterStart = j + 1; // reset cluster
+        streak = 0;
+      }
+    }
+  }
+  clusters = SamplePartition(sample_, indicesCollection);
+  return peaks;
+}
+
+Graph SamplePartition::draw(const Scalar threshold) const
+{
+  Graph graph(sample_.getDescription()[0] + " clusters", "Index", sample_.getDescription()[0], true, "");
+  const UnsignedInteger size = sample_.getSize();
+  Sample indexSample(size, 1);
+  for (UnsignedInteger i = 0; i < size; ++ i)
+    indexSample(i, 0) = i;
+
+  Scalar yMin = sample_.getMin()[0];
+  Scalar yMax = sample_.getMax()[0];
+  const Scalar delta = yMax - yMin;
+  yMin -= 0.05 * delta;
+  yMax += 0.05 * delta;
+
+  // loop over partitions
+  Sample peaks(0, 1);
+  Sample peakIndices(0, 1);
+  for (UnsignedInteger i = 0; i < indicesCollection_.getSize(); ++ i)
+  {
+    const UnsignedInteger partitionStart = indicesCollection_[i][0];
+    const UnsignedInteger partitionEnd = indicesCollection_[i][1];
+    Scalar peak = threshold;
+    Scalar peakIndex = 0;
+    for (UnsignedInteger j = partitionStart; j < partitionEnd; ++ j)
+    {
+      if (sample_(j, 0) > peak)
+      {
+        peak = sample_(j, 0);
+        peakIndex = j;
+      }
+    }
+    peaks.add(Point({peak}));
+    peakIndices.add(Point({peakIndex}));
+
+    // cluster begin vertical lines
+    Point dataX = {partitionStart - 0.5, partitionStart - 0.5};
+    const Point dataY = {yMin, yMax};
+    Curve vLine = Curve(dataX, dataY);
+    vLine.setColor("gray");
+    vLine.setLineStyle("dashed");
+    graph.add(vLine);
+    dataX = {partitionEnd + 0.5, partitionEnd + 0.5};
+    vLine = Curve(dataX, dataY);
+    vLine.setColor("gray");
+    vLine.setLineStyle("dashed");
+    graph.add(vLine);
+
+    // cluster shade
+    const Point rectX = {partitionStart - 0.5, partitionEnd + 0.5, partitionEnd + 0.5, partitionStart - 0.5};
+    const Point rectY = {yMin, yMin, yMax, yMax};
+    Polygon rect(rectX, rectY);
+    rect.setColor("lightgray");
+    graph.add(rect);
+  }
+
+  // add threshold horizontal line
+  const Point dataX = {-0.5, static_cast<Scalar>(size) + 0.5};
+  const Point dataY(2, threshold);
+  Curve hLine(dataX, dataY);
+  hLine.setColor("black");
+  hLine.setLineStyle("dashed");
+  graph.add(hLine);
+  Text elt(Point({1.0}), Point({threshold}), {"threshold"}, "top");
+  elt.setColor("black");
+  graph.add(elt);
+
+  // add cloud
+  Cloud allCloud(indexSample, sample_);
+  allCloud.setPointStyle(".");
+  allCloud.setColor("black");
```

Comment:
No we dont want colors here

Reply:
This is not managed by the color, but by the legend associated with each set of points. I maintain that no color has to be set here because the default palette will provide a color.

Reply:
I dont want the default palette color or any color, I want a cloud of black points